packages/deepbook_margin/sources/margin_manager.move

```diff
@@ -143,6 +148,157 @@ public struct WithdrawCollateralEvent has copy, drop {
+    conditional_order_id: u64,
+    clock: &Clock,
+    ctx: &TxContext,
+) {
+    self.validate_owner(ctx);
+    let manager_id = self.id();
+    self.take_profit_stop_loss.cancel_conditional_order(manager_id, conditional_order_id, clock);
+}
+
+/// Execute conditional orders and return the order infos.
+/// This is a permissionless function that can be called by anyone.
+public fun execute_conditional_orders<BaseAsset, QuoteAsset>(
+    self: &mut MarginManager<BaseAsset, QuoteAsset>,
+    pool: &mut Pool<BaseAsset, QuoteAsset>,
+    base_price_info_object: &PriceInfoObject,
+    quote_price_info_object: &PriceInfoObject,
+    registry: &MarginRegistry,
+    max_orders_to_execute: u64,
+    clock: &Clock,
+    ctx: &TxContext,
+): vector<OrderInfo> {
+    assert!(pool.id() == self.deepbook_pool(), EIncorrectDeepBookPool);
+    let current_price = calculate_price<BaseAsset, QuoteAsset>(
+        registry,
+        base_price_info_object,
+        quote_price_info_object,
+        clock,
+    );
+
+    let mut order_infos = vector[];
+    let mut executed_ids = vector[];
+    let mut expired_ids = vector[];
+    let mut insufficient_funds_ids = vector[];
+
+    // Collect orders to process (to avoid borrow conflicts)
+    let mut orders_to_process = vector[];
+
+    // Collect trigger_below orders (sorted high to low)
+    let mut i = 0;
+    while (i < self.take_profit_stop_loss.trigger_below().length()) {
+        let conditional_order = &self.take_profit_stop_loss.trigger_below()[i];
+
+        // Break early if price doesn't trigger
+        if (current_price >= conditional_order.condition().trigger_price()) {
+            break
+        };
+
+        orders_to_process.push_back(*conditional_order);
+        i = i + 1;
+    };
+
+    // Collect trigger_above orders (sorted low to high)
+    i = 0;
+    while (i < self.take_profit_stop_loss.trigger_above().length()) {
+        let conditional_order = &self.take_profit_stop_loss.trigger_above()[i];
+
+        // Break early if price doesn't trigger
+        if (current_price <= conditional_order.condition().trigger_price()) {
+            break
+        };
+
+        orders_to_process.push_back(*conditional_order);
+        i = i + 1;
+    };
+
+    // Process collected orders
+    self.process_collected_orders(
+        pool,
+        registry,
+        orders_to_process,
+        &mut order_infos,
+        &mut executed_ids,
+        &mut expired_ids,
+        &mut insufficient_funds_ids,
+        max_orders_to_execute,
+        clock,
+        ctx,
+    );
+
+    let manager_id = self.id();
+    let pool_id = pool.id();
+
+    insufficient_funds_ids.do!(|id| {
+        self.take_profit_stop_loss.emit_insufficient_funds_event(manager_id, id, clock);
+    });
+
+    let mut cancelled_ids = expired_ids;
+    cancelled_ids.append(insufficient_funds_ids);
+    cancelled_ids.do!(|id| {
+        self.take_profit_stop_loss.cancel_conditional_order(manager_id, id, clock);
```

Comment:
nit: maybe note here that cancel orders will include the ones with insufficient funds